### DISCUSSION TITLE
When will Install-DbaInstance support SQL Server 2022?

### DISCUSSION THREAD

**Discussion by 0x7FFFFFFFFFFFFFFF** [dataplat/dbatools#8569] [General] [answered]
Today I was trying to install a SQL Server 2022 instance for my lab. But I found that `Install-DbaInstance`'s `Version` parameter doesn't support SQL Server 2022. Do we have a plan to support it? If so, when will it happen? Because installing an instance with `dbatools` is so easy, and I'm already used to it.

**Comment by niphlod** [comment, marked as answer]
generally only when RTM is available. in RC phase nobody still knows if it'll work or not.

**Reply by 0x7FFFFFFFFFFFFFFF** [reply]
Thanks!

**Comment by m1chael-d** [comment]
What about now? It would be nice to be able to override the version check and install newer versions not yet supported by dbatools.

**Reply by potatoqualitee** [reply]
It's supported in the dev branch which is dbatools 2.0, but not in 1.1.x :/ You can `Install-Module dbatools.library -AllowPreRelease` then grab https://dbatools.io/devzip, put that in your `$env:PSModulePath` and it'll work.

**Reply by m1chael-d** [reply]
thanks for the quick reply, i am getting this error;

![image](https://user-images.githubusercontent.com/32747976/209193319-a73ba096-e2eb-4826-ae1a-b6e317b2f337.png)

**Reply by m1chael-d** [reply]
It would be great to override the version check and provide additional parameters like, EULA etc

**Reply by potatoqualitee** [reply]
ahh that may be too old of a version of powershell to do the allowprerelease, good to know. you could download it manually from powershellgallery.com but it's not in a solid enough state for your current version, sorry :/ we will continue to support PS v3 and v4 but it takes some additional work. as for your other request, you'd have to file an issue with details on what you'd like to see implemented. the request seems reasonable to me!
